From 828e8a3e693cd189f730aa80ad85c02f409fd9d1 Mon Sep 17 00:00:00 2001
From: mshankar <mshankar@maxlinear.com>
Date: Wed, 28 Aug 2024 18:39:49 +0800
Subject: [PATCH] WLANRTSYS-82318 Populate the VHT capability for 5G STA during
 WDS connection

Issue: Seeing low throughput numbers in 2.4/5G WDS STR
RC: We miss to update the VHT cap info for 5G sta under specific scenario due to which
the staDB of this 5G sta is filled with the incorrect/minimum maxMpdu limit. This
maxMpdu limit is configured based on the vht capabilities that we send. Under specific
scenario this cap info is all 0s leading to incorrect maxMpdu limit for that station.

Fix: Taken care to update the vht cap info of the station in all the scenarios.

FIXUP to WLANRTSYS-63677 ML WDS configuration and association
---
 src/ap/mxl_wds.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/ap/mxl_wds.c b/src/ap/mxl_wds.c
index 7be3ed5ea..eae018043 100644
--- a/src/ap/mxl_wds.c
+++ b/src/ap/mxl_wds.c
@@ -433,6 +433,11 @@ void mxl_wds_event_connect(struct hostapd_data *hapd,
 			hostapd_get_vht_capab(hapd->mxl_data.sibling_hapd, sta->mxl_sta_info.linked_sta->vht_capabilities, &vht_cap[0]);
 #endif /* CONFIG_IEEE80211BE_MXL_MLO */
 	}
+#ifdef CONFIG_IEEE80211BE_MXL_MLO
+	else if (is_both_ml_sta_assoc_received && sta->mxl_sta_info.linked_sta) {
+		hostapd_get_vht_capab(hapd->mxl_data.sibling_hapd, sta->mxl_sta_info.linked_sta->vht_capabilities, &vht_cap[0]);
+	}
+#endif
 #endif /* CONFIG_IEEE80211AC */
 
 #ifdef CONFIG_IEEE80211AX
-- 
2.43.0

